config/vlan.py

```diff
@@ -74,57 +99,74 @@ def delete_db_entry(entry_name, db_connector, db_name):
-
-        keys = [ (k, v) for k, v in db.cfgdb.get_table('VLAN_MEMBER') if k == 'Vlan{}'.format(vid) ]
-
-        if keys: # TODO: MISSING CONSTRAINT IN YANG MODEL
-            ctx.fail("VLAN ID {} can not be removed. First remove all members assigned to this VLAN.".format(vid))
-
-        vxlan_table = db.cfgdb.get_table('VXLAN_TUNNEL_MAP')
-        for vxmap_key, vxmap_data in vxlan_table.items():
-            if vxmap_data['vlan'] == 'Vlan{}'.format(vid):
-                ctx.fail("vlan: {} can not be removed. First remove vxlan mapping '{}' assigned to VLAN".format(vid, '|'.join(vxmap_key)) )
+        for vid in vid_list:
+            log.log_info("'vlan del {}' executing...".format(vid))
+            
+            if not clicommon.is_vlanid_in_range(vid):
+                ctx.fail("Invalid VLAN ID {} (2-4094)".format(vid))
+            
+            #Multiple VLANs needs to be referenced
+            vlan = 'Vlan{}'.format(vid)
+
+            #Multiple VLANs needs to be checked
+            if no_restart_dhcp_relay:
+                if is_dhcpv6_relay_config_exist(db, vlan):
+                    ctx.fail("Can't delete {} because related DHCPv6 Relay config is exist".format(vlan))
+
+            if clicommon.check_if_vlanid_exist(db.cfgdb, vlan) == False:
+                log.log_info("{} does not exist".format(vlan))
+                ctx.fail("{} does not exist, Aborting!!!".format(vlan))
+
+            intf_table = db.cfgdb.get_table('VLAN_INTERFACE')
+            for intf_key in intf_table:
+                if ((type(intf_key) is str and intf_key == 'Vlan{}'.format(vid)) or  # TODO: MISSING CONSTRAINT IN YANG MODEL
+                        (type(intf_key) is tuple and intf_key[0] == 'Vlan{}'.format(vid))):
+                    ctx.fail("{} can not be removed. First remove IP addresses assigned to this VLAN".format(vlan))
+
+            keys = [(k, v) for k, v in db.cfgdb.get_table('VLAN_MEMBER') if k == 'Vlan{}'.format(vid)]
+
+            if keys:  # TODO: MISSING CONSTRAINT IN YANG MODEL
+                ctx.fail("VLAN ID {} can not be removed. First remove all members assigned to this VLAN.".format(vid))
+
+            vxlan_table = db.cfgdb.get_table('VXLAN_TUNNEL_MAP')
+            for vxmap_key, vxmap_data in vxlan_table.items():
+                if vxmap_data['vlan'] == 'Vlan{}'.format(vid):
+                    ctx.fail("vlan: {} can not be removed. First remove vxlan mapping '{}' assigned to VLAN".format(vid, '|'.join(vxmap_key))) 
+        
+            # set dhcpv4_relay table
+            set_dhcp_relay_table('VLAN', config_db, vlan, None)
 
-    # set dhcpv4_relay table
-    set_dhcp_relay_table('VLAN', config_db, vlan, None)
+            if not no_restart_dhcp_relay and is_dhcpv6_relay_config_exist(db, vlan):
+                # set dhcpv6_relay table
+                set_dhcp_relay_table('DHCP_RELAY', config_db, vlan, None)
```

Comment:
same as above on `set_dhcp_relay_table`

Reply:
This set_dhcp_relay_table is not in scope of our HLD and we have not modified any of this exisiting behavior